stubs/inertia-react-ts/resources/js/types/index.d.ts

```diff
@@ -0,0 +1,12 @@
+export interface User {
+    id: number;
+    name: string;
+    email: string;
+    email_verified_at: string;
+}
+
+export type PageProps<T extends Record<string, unknown> = {}> = T & {
```

Comment:
In this case I don't think it makes much difference because it's just being used in an intersection type, but I've changed it to `Record<string, never>` anyway if that helps with eslint. It's just there as a default so that people don't always need to specify the generic argument in cases where they only care about the shared props.

As for `email_verified_at`, this key comes from Eloquent via Inertia.